docs/pages/mintlist/[mintlistAddress].tsx

```diff
@@ -244,5 +279,274 @@ function useMintlist({
+                      {values.nfts.length <= MAX_NFTS_PER_BATCH &&
+                        values.nfts.length < availableToUpload && (
+                          <button
+                            type="button"
+                            className="add-nft-button animated"
+                            onClick={() =>
+                              insert(values.nfts.length, {
+                                name: "",
+                                image: "",
+                              })
+                            }
+                          >
+                            <PlusIcon
+                              style={{
+                                width: "2rem",
+                                height: "2rem",
+                              }}
+                            />
+                          </button>
+                        )}
+                    </>
+                  )}
+                </FieldArray>
+              </div>
+              <div className="mt-4 flex-center spread">
+                <LuxSubmitButton
+                  label={`Upload ${values.nfts.length} NFT${
+                    values.nfts.length !== 1 ? "s" : ""
+                  }`}
+                  rounded
+                  color="brand"
+                  disabled={!(isValid && values.nfts.every((nft) => nft.image))}
+                />
+                <LuxButton
+                  label="Disconnect Wallet"
+                  onClick={onSignOut}
+                  color="secondary"
+                  size="small"
+                  variant="link"
+                />
+              </div>
+            </Form>
+          )}
+        </Formik>
+      </InteractiveWell>
+      <style jsx>{`
+        .grid {
+          display: grid;
+          grid-template-columns: repeat(3, 1fr);
+          column-gap: 1rem;
+          row-gap: 2rem;
+        }
+
+        .add-nft-button {
+          border: 1px solid var(--primary-border-color);
+          border-radius: var(--border-radius);
+          background-color: var(--faint-gray);
+          // Make sure the button doesn't collapse when the grid is empty.
+          min-height: 11.8rem;
+        }
+        .add-nft-button:hover {
+          background-color: var(--pale-gray);
+        }
+      `}</style>
+    </>
+  );
+}
+
+function NftsGrid({ mintInfos }: { mintInfos: NftokenTypes.MintInfo[] }) {
+  const { data: metadataMap } = useMintInfosMetadata(mintInfos);
+
+  if (!mintInfos.length) {
+    return <p>No NFTs have been uploaded to this mintlist yet.</p>;
+  }
+
+  const mintInfosWithMetadata = mintInfos
+    .filter(({ metadata_url }) => metadataMap.get(metadata_url))
+    .map((mintInfo) => ({
+      ...mintInfo,
+      metadata: metadataMap.get(mintInfo.metadata_url)!,
+    }));
+
+  return (
+    <>
+      <div className="grid">
+        {mintInfosWithMetadata.map((mintInfo) => (
+          <figure className="nft-card" key={mintInfo.metadata_url}>
+            <img
```

Comment:
We should have an `<SquareImage />` component cc @benborgers

Reply:
Yes! Just landed in #162 last night.

Reply:
Nice added this to the TODO: https://linear.app/luma/issue/L-5251/polish-the-mintlist-functionality